cypress/integration/channel/message_edit_post_with_attachment_spec.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 3] */
+
+describe('MM-13697 Edit Post with attachment', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('Pasted text should be pasted where the cursor is', () => {
+        // 2. Got to a test channel on the side bar
+        cy.get('#sidebarItem_town-square').should('be.visible').click({force: true});
+
+        // * Validate if the channel has been opened
+        cy.url().should('include', '/ad-1/channels/town-square');
+
+        // 3. Attach image
+        cy.uploadFile('#fileUploadButton input', '../fixtures/mattermost-icon.png', 'image/png');
+
+        // 4. Type 'This is sample text'
+        cy.get('#post_textbox').type('This is sample text').should('contain', 'This is sample text');
+
+        // 5. Submit post
+        cy.get('#create_post').submit();
```

Comment:
Getting last post ID in step 6 will sometime yield to temporary `pending_post_id`.  I'd suggest to add wait time here.

```suggestion
        cy.get('#create_post').submit();
        cy.wait(500); // eslint-disable-line
```